src/main/java/uk/gov/justice/digital/zone/RawZone.java

```diff
@@ -40,58 +49,69 @@ public RawZone(JobArguments jobArguments,
     }
 
     @Override
-    public Dataset<Row> process(SparkSession spark, Dataset<Row> dataFrame, Row table) throws DataStorageException {
+    public Dataset<Row> processLoad(SparkSession spark, Dataset<Row> dataFrame, Row table) throws DataStorageException {
+
+        val sortedLoadRecords = dataFrame
+                .filter(col(OPERATION).equalTo(Load.getName()))
+                .orderBy(col(TIMESTAMP));
+
+        return process(spark, sortedLoadRecords, table);
+    }
+
+    @Override
+    public Dataset<Row> processCDC(SparkSession spark, Dataset<Row> dataFrame, Row table) throws DataStorageException {
+
+        val sortedCDCRecords = dataFrame
+                .filter(col(OPERATION).isInCollection(cdcOperationsSet))
+                .repartition(1)
+                .orderBy(col(TIMESTAMP));
 
-        final long count = dataFrame.count(); // this is expensive. Requires a shuffle.
+        return process(spark, sortedCDCRecords, table);
+    }
 
+    private Dataset<Row> process(SparkSession spark, Dataset<Row> records, Row row) throws DataStorageException {
+        val count = records.count();
         val startTime = System.currentTimeMillis();
 
-        String rowSource = table.getAs(SOURCE);
-        String rowTable = table.getAs(TABLE);
-        String rowOperation = table.getAs(OPERATION);
+        String rowSource = row.getAs(SOURCE);
+        String rowTable = row.getAs(TABLE);
 
-        logger.info("Processing batch with {} records for source: {} table: {}",
+        logger.info("Processing raw batch with {} records for source: {} table: {}",
```

Comment:
This is fine, but bear in mind that the class is also in the log output so we can see which zone it is already. I can show you some log examples if needed.